support/manpage/places.1
.\"t
.\" Automatically generated by Pandoc 2.9.2.1
.\"
.TH "places" "1" "2024-06-21" "places 0.1.1" "User Manual"
.hy
.SH NAME
.PP
places - Get paths for common places
.SH SYNOPSIS
.PP
\f[B]places\f[R] [\f[B]OPTIONS\f[R]] <\f[B]location\f[R]>
[\f[B]pathel\f[R]]\&...
.SH DESCRIPTION
.PP
A handy tool for your scripts to get correct path names from known
locations or XDG directories.
.SH ARGUMENTS
.TP
\f[B]\f[CB]<location>\f[B]\f[R]
The location.
See table bellow for a list of known locations.
.TP
\f[B]\f[CB][pathel]...\f[B]\f[R]
Optional list of path elements to append
.SH OPTIONS
.TP
\f[B]\f[CB]-a\f[B]\f[R], \f[B]\f[CB]--app\f[B]\f[R]
Optional program to get paths
.TP
\f[B]\f[CB]-h\f[B]\f[R], \f[B]\f[CB]--help\f[B]\f[R]
Print help (see a summary with \f[C]-h\f[R])
.TP
\f[B]\f[CB]-V\f[B]\f[R], \f[B]\f[CB]-version\f[B]\f[R]
Print version
.SH LOCATION TABLE
.PP
The following is the list of known locations
.SS \f[B]XDG\f[R] locations
.PP
.TS
tab(@);
l l l.
T{
name
T}@T{
environment var
T}@T{
example
T}
_
T{
\f[B]home\f[R]
T}@T{
$HOME
T}@T{
\f[I]/home/alice\f[R]
T}
T{
\f[B]desktop\f[R]
T}@T{
$XDG_DESKTOP_DIR
T}@T{
\f[I]/home/alice/Desktop\f[R]
T}
T{
\f[B]downloads\f[R]
T}@T{
$XDG_DOWNLOAD_DIR
T}@T{
\f[I]/home/alice/Downloads\f[R]
T}
T{
\f[B]templates\f[R]
T}@T{
$XDG_TEMPLATES_DIR
T}@T{
\f[I]/home/alice/Templates\f[R]
T}
T{
\f[B]documents\f[R]
T}@T{
$XDG_DOCUMENTS_DIR
T}@T{
\f[I]/home/alice/Documents\f[R]
T}
T{
\f[B]pictures\f[R]
T}@T{
$XDG_PICTURES_DIR
T}@T{
\f[I]/home/alice/Pictures\f[R]
T}
T{
\f[B]videos\f[R]
T}@T{
$XDG_VIDEOS_DIR
T}@T{
\f[I]/home/alice/Videos\f[R]
T}
T{
\f[B]music\f[R]
T}@T{
$XDG_MUSIC_DIR
T}@T{
\f[I]/home/alice/Music\f[R]
T}
T{
\f[B]public\f[R]
T}@T{
$XDG_PUBLICSHARE_DIR
T}@T{
\f[I]/home/alice/Public\f[R]
T}
.TE
.SS Common locations
.PP
.TS
tab(@);
l l l.
T{
name
T}@T{
environment var
T}@T{
example
T}
_
T{
\f[B]config\f[R]
T}@T{
$XDG_CONFIG_HOME
T}@T{
\f[I]/home/alice/.config\f[R]
T}
T{
\f[B]bin\f[R]
T}@T{
$XDG_BIN_HOME
T}@T{
\f[I]/home/alice/.local/bin\f[R]
T}
T{
\f[B]cache\f[R]
T}@T{
$XDG_CACHE_HOME
T}@T{
\f[I]/home/alice/.cache\f[R]
T}
T{
\f[B]state\f[R]
T}@T{
$XDG_STATE_HOME
T}@T{
\f[I]/home/alice/.state\f[R]
T}
T{
\f[B]data\f[R]
T}@T{
$XDG_DATA_HOME
T}@T{
\f[I]/home/alice/.local/share\f[R]
T}
.TE
.SS Other locations
.PP
Note the tool always return a full path, not paths starting with
\f[I]`\[ti]/'\f[R].
.PP
.TS
tab(@);
l l.
T{
name
T}@T{
example
T}
_
T{
\f[B]autostart\f[R]
T}@T{
\f[I]\[ti]/.config/autostart\f[R]
T}
T{
\f[B]fonts\f[R]
T}@T{
\f[I]\[ti]/.local/share/fonts\f[R]
T}
T{
\f[B]menus\f[R]
T}@T{
\f[I]\[ti]/.local/share/applications\f[R]
T}
T{
\f[B]backgrounds\f[R]
T}@T{
\f[I]\[ti]/.local/share/backgrounds\f[R]
T}
.TE
.SH EXAMPLES
.PP
But you can do much more with it
.SS * Get a path like: \f[I]/home/alice/.config/gizmo/theme/colors.txt\f[R]
.PP
\f[B]$\f[R] places -a gizmo config theme colors.txt
.PD 0
.P
.PD
\f[B]$\f[R] places config gizmo theme colors.txt
.SS * List all config files for the \f[I]`gizmo'\f[R] program
.PP
\f[B]$\f[R] lsd $(places \[en]app=gizmo data)
.PD 0
.P
.PD
\f[B]$\f[R] places \[en]app=gizmo data | xargs lsd
.SS * Copy a config file
.PP
\f[B]$\f[R] cp gizmo_config.toml $(places -a gizmo config)
.SS * Get a folder inside \f[C]\[aq]Downloads dir\[aq]\f[R], even if it is localized.
.PP
\f[B]$\f[R] places downloads Software
.PP
This returns something like \f[I]/home/alice/Descargas/Software\f[R]
.SH ENVIRONMENT
.PP
The list of environment variables that affect the operation is listed on
the tables above.
.PP
The \f[B]XDG\f[R] localized names is defined in the file
\f[I]/etc/xdg/user-dirs.defaults\f[R] and for each user in the file
\f[I]$(XDG_CONFIG_HOME)/user-dirs.dirs\f[R].
.SH BUGS
.PP
See GitHub Issues:
\f[I]https://github.com/jtsagata/places-cli/issues\f[R]
.SH SEE ALSO
.PP
\f[B]xdg-user-dir(1)\f[R], \f[B]user-dirs.dirs(5)\f[R]


CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## v0.1.2
### Added
- Shorter readme for crates.io
- Create debian packages with cargo-deb
### Changed
- Typos and small fixes to README and manpages
### Fixed
- The manage is now generated from markdown using pandoc

## v0.1.1
### Added
### Changed
- Add badges to Readme.md

## v0.1
### Added
- Basic functionality
- Github deployment
- Bump softprops/action-gh-release from 1 to 2
- Bump clap from 4.5.6 to 4.5.7




support/completions/places.bash
_places() {
    local i cur prev opts cmd
    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    prev="${COMP_WORDS[COMP_CWORD-1]}"
    cmd=""
    opts=""

    for i in ${COMP_WORDS[@]}
    do
        case "${cmd},${i}" in
            ",$1")
                cmd="places"
                ;;
            *)
                ;;
        esac
    done

    case "${cmd}" in
        places)
            opts="-a -h -V --app --help --version home desktop downloads templates documents pictures videos music public config bin cache state data autostart fonts menus backgrounds icons [pathel]..."
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --app)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -a)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
    esac
}

if [[ "${BASH_VERSINFO[0]}" -eq 4 && "${BASH_VERSINFO[1]}" -ge 4 || "${BASH_VERSINFO[0]}" -gt 4 ]]; then
    complete -F _places -o nosort -o bashdefault -o default places
else
    complete -F _places -o bashdefault -o default places
fi
